File: AnnotationStudy/app/src/main/java/com/example/annotationstudy/MainActivity.java
package com.example.annotationstudy;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.tv)
    TextView tv;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        BindViewUtils.bind(this);
        tv.setText("Hello, 2021!");
    }
}

File: CustomAttributesStudy/app/src/main/java/com/example/customattributesstudy/MainActivity.kt
package com.example.customattributesstudy

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Choreographer
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        window.decorView.setOnFocusChangeListener { v, hasFocus ->

        }
        window.decorView.requestFocus()
        Choreographer.getInstance().postFrameCallback {  }
    }
}